### ISSUE TITLE
java.lang.NoSuchMethodError when using lettuce 6.x

### ISSUE THREAD

**Issue by xenomachina** [mokies/ratelimitj#49]
It looks like lettuce 6.x introduced a binary incompatibility that makes ratelimitj break at runtime.

The breakage we're running into is that `RedisScriptLoader` calls `RedisScriptingReactiveCommands.scriptLoad`, and this changed from being a single method with the signature `Mono<String> scriptLoad(V script)` to a pair of methods, `Mono<String> scriptLoad(String script)` and `Mono<String> scriptLoad(byte[] script)`.

This change appears to be source compatible with ratelimitj. I tried changing ratelimitj's dependency on lettuce from `io.lettuce:lettuce-core:5.3.4.RELEASE` to `io.lettuce:lettuce-core:6.0.2.RELEASE`, and `./gradlew ratelimitj-redis:check` builds and passes with the new version of lettuce.

**Comment by Akaame** [comment]
I think this has been already covered by

From [this](https://github.com/mokies/ratelimitj/commit/05a361a2fc5fe63a9a7ba18b231f85ba19d0f6c7#diff-c202195541080723c412ab8663a363f36856cccc3d14fb763dedc9a3431bd812) to [this](https://github.com/mokies/ratelimitj/commit/eb58d3d575c8ecec8cab40212a810c9acedfe110#diff-c202195541080723c412ab8663a363f36856cccc3d14fb763dedc9a3431bd812) and is available from 0.7.0 release on.

**Comment by sycz** [comment]
I have the same problem. 0.7.0 Incompatible with lettuce-core:5.3.7.

```
es.moki.ratelimitj.redis.request.RedisScriptLoader.loadScript()

java.lang.NoSuchMethodError: io.lettuce.core.api.reactive.RedisScriptingReactiveCommands.scriptLoad(Ljava/lang/String;)Lreactor/core/publisher/Mono;
```

I compared the two versions(ratelimitj-redis 0.7.0-RC3 / 0.7.0) and found that they were basically the same. Except this: 
```
.retry(1, RedisSlidingWindowRequestRateLimiter::startWithNoScriptError)

.retryWhen(Retry.max(1).filter(RedisSlidingWindowRequestRateLimiter::startWithNoScriptError))
```

ratelimitj-core doesn't change anything.

But if you run it(ratelimitj-redis:0.7.0) in source code, it works.

I think it(NoSuchMethodError) should not be affected by dependencies.

I wonder what caused it.

**Comment by xenomachina** [comment]
The incompatibility with lettuce-core 6.x was fixed in f37eb875addb50e1f571b7c4b6f2d5d41802c46e, when the dependency on lettuce-core was upgraded from 5.3.4 to 6.0.2, so this issue is resolved as reported.

**However, the fix means that users of ratelimitj need to also upgrade lettuce-core 6.x.**

@sycz I don't know about now, but back when I reported this issue, the incompatibility wouldn't bite you if you built ratelimitj yourself, because the change that was made to lettuce-core was still source-compatible with ratelimitj, just not _binary_ compatible with anything that used the `scriptLoad` method from the "wrong" version. Can you upgrade your lettuce-core dependency to 6.x? (The current release is 6.1.8.)
